src/UCTSearch.h

```diff
@@ -102,4 +109,32 @@ class UCTWorker {
     UCTNode* m_root;
 };
 
+/// LimitsType struct stores information sent by GUI about available time to
+/// search the current move, maximum depth/time, or if we are in analysis mode.
+
+struct LimitsType {
+
+    LimitsType() { // Init explicitly due to broken value-initialization of non POD in MSVC
+        nodes = time[WHITE] = time[BLACK] = inc[WHITE] = inc[BLACK] =
+        npmsec = movestogo = depth = movetime = mate = perft = infinite = 0;
+        startTime = now();
+    }
+
+    std::atomic<bool> dynamic_controls_set() const {
+        return {(time[WHITE] | time[BLACK] | inc[WHITE] | inc[BLACK] | npmsec | movestogo) != 0};
+    }
+
+    std::atomic<bool> use_time_management() const {
```

Comment:
I don't think these need to be atomic bool - the limits don't change after they are initialized, and the function is const.

Reply:
Good point, I think I left that there after some experimentation, I'll change that now.